wagtail/api/v2/pagination.py

```diff
@@ -33,7 +33,7 @@ def paginate_queryset(self, queryset, request, view=None):
         stop = offset + limit
 
         self.view = view
-        self.total_count = queryset.count()
+        self.total_count = len(queryset) if isinstance(queryset, list) else queryset.count()
```

Comment:
Could you make it more general by reversing this condition?
```python
self.total_count = queryset.count() if isinstance(queryset, QuerySet) else len(queryset)
```

Just in case a search backend returns a tuple or some kind of cursor, for example.